bouncer/tests/btc_vault_swap.ts

```diff
@@ -0,0 +1,120 @@
+import assert from 'assert';
+import { ExecutableTest } from '../shared/executable_test';
+import { BTC_ENDPOINT, selectInputs, waitForBtcTransaction, btcClient } from '../shared/send_btc';
+import {
+  amountToFineAmount,
+  Asset,
+  assetDecimals,
+  btcClientMutex,
+  createStateChainKeypair,
+  newAddress,
+  observeBalanceIncrease,
+} from '../shared/utils';
+import { getChainflipApi, observeEvent } from '../shared/utils/substrate';
+import { getBalance } from '../shared/get_balance';
+
+/* eslint-disable @typescript-eslint/no-use-before-define */
+export const testBtcVaultSwap = new ExecutableTest('Btc-Vault-Swap', main, 100);
+
+interface VaultSwapDetails {
+  chain: string;
+  nulldata_utxo: string;
+  deposit_address: string;
+}
+
+async function buildAndSendBtcVaultSwap(
+  depositAmountBtc: number,
+  brokerUri: string,
+  destinationAsset: Asset,
+  destinationAddress: string,
+  refundAddress: string,
+) {
+  await using chainflip = await getChainflipApi();
+
+  const broker = createStateChainKeypair(brokerUri);
+  testBtcVaultSwap.debugLog('Broker:', broker.address);
+  testBtcVaultSwap.debugLog(`Btc endpoint is set to`, BTC_ENDPOINT);
+
+  const feeBtc = 0.00001;
+  const { inputs, change } = await selectInputs(Number(depositAmountBtc) + feeBtc);
+
+  const vaultSwapDetails = (await chainflip.rpc(
+    `cf_get_vault_swap_details`,
+    broker.address,
+    'BTC', // source_asset
+    destinationAsset.toUpperCase(),
+    destinationAddress,
+    0, // broker_commission
+    0, // min_output_amount
+    0, // retry_duration
+  )) as unknown as VaultSwapDetails;
+
+  assert.strictEqual(vaultSwapDetails.chain, 'Bitcoin');
+  testBtcVaultSwap.debugLog('nulldata_utxo:', vaultSwapDetails.nulldata_utxo);
+
+  // The `createRawTransaction` function will add the op codes, so we have to remove them here.
+  const nullDataWithoutOpCodes = vaultSwapDetails.nulldata_utxo.replace('0x', '').substring(4);
+
+  const outputs = [
+    {
+      [vaultSwapDetails.deposit_address]: depositAmountBtc,
+    },
+    {
+      data: nullDataWithoutOpCodes,
+    },
+    {
+      [refundAddress]: change,
+    },
+  ];
+
+  const rawTx = await btcClient.createRawTransaction(inputs, outputs, 0, false);
+  const signedTx = await btcClient.signRawTransactionWithWallet(rawTx);
+  const txid = await btcClientMutex.runExclusive(async () =>
+    btcClient.sendRawTransaction(signedTx.hex),
+  );
+  if (!txid) {
+    throw new Error('Broadcast failed');
+  } else {
+    testBtcVaultSwap.log('Broadcast successful, txid:', txid);
+  }
```

Comment:
nit, early return makes it a little easier to read
```suggestion
if (!txid) throw new Error('Broadcast failed');
testBtcVaultSwap.log('Broadcast successful, txid:', txid);
```